tools/WindowStylesReportTool/WindowStylesReportTool.cpp

```diff
@@ -0,0 +1,414 @@
+#include "pch.h"
+#include "WindowStylesReportTool.h"
+
+#include <dwmapi.h>
+#include <shlobj.h>
+
+#include <filesystem>
+#include <fstream>
+#include <map>
+
+inline std::optional<std::wstring> get_last_error_message(const DWORD dw)
+{
+    std::optional<std::wstring> message;
+    try
+    {
+        const auto msg = std::system_category().message(dw);
+        message.emplace(begin(msg), end(msg));
+    }
+    catch (...)
+    {
+    }
+    return message;
+}
+
+inline std::wstring get_last_error_or_default(const DWORD dw)
+{
+    auto message = get_last_error_message(dw);
+    return message.has_value() ? message.value() : L"";
+}
+
+std::filesystem::path get_desktop_path()
+{
+    wchar_t* p;
+    if (S_OK != SHGetKnownFolderPath(FOLDERID_Desktop, 0, NULL, &p)) return "";
+
+    std::filesystem::path result = p;
+    CoTaskMemFree(p);
+
+    return result;
+}
+
+// Get the executable path or module name for modern apps
+inline std::wstring get_process_path(DWORD pid) noexcept
+{
+    auto process = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, TRUE, pid);
```

Comment:
```suggestion
    auto process = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, TRUE, pid);
```

We should use `PROCESS_QUERY_LIMITED_INFORMATION (0x1000)`. Otherwise name is empty on elevated processes.

Reply:
Should it help? I don't see any difference when trying to get the name of TaskManager. 
According to docs, `The PROCESS_QUERY_LIMITED_INFORMATION right was introduced to provide access to a subset of the information available through PROCESS_QUERY_INFORMATION`, so we won't receive more info with this right. 

We can start the tool elevated to get the name of the elevated process :)

Reply:
> Should it help? I don't see any difference when trying to get the name of TaskManager. 
> According to docs, `The PROCESS_QUERY_LIMITED_INFORMATION right was introduced to provide access to a subset of the information available through PROCESS_QUERY_INFORMATION`, so we won't receive more info with this right. 

It won't receive more. It receive lesse and therefore the query isn't blocked anymore with E_ACCESS_DINIED. It's the same fix we did in WW plugin.

> We can start the tool elevated to get the name of the elevated process :)

Sure. But with the fix it's easier and you might have a problem (access the window if original user, different desktop folder) if you need to run as different user to get admin rights.